src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -95,6 +95,20 @@ bool EnableUniqueJobIds = true;
 static List *OperatorCache = NIL;
 
 
+/* context passed down in AddAnyValueAggregates mutator */
+typedef struct AddAnyValueAggregatesContext
+{
+	/* SortGroupClauses corresponding to the GROUP BY clause */
+	List *groupClauseList;
+
+	/* TargetEntry's to which the GROUP BY clauses refer */
+	List *groupByTargetEntryList;
+
+	/* true if there are expressions in the GROUP BY target entries */
```

Comment:
minor: I was initially confused a little about `haveNonVarGrouping `. So, maybe mention that this is for preventing unnecessary checks during the processiing?